lucene/core/src/java/org/apache/lucene/index/SegmentInfos.java

```diff
@@ -440,7 +440,7 @@ public static final SegmentInfos readCommit(Directory directory, ChecksumIndexIn
       if (format >= VERSION_70) { // oldest supported version
         CodecUtil.checkFooter(input, priorE);
       } else {
-        throw IOUtils.rethrowAlways(priorE);
```

Comment:
Leave this as it was (with throw ...) - don't know whether IntelliJ is smart enough to detect this method always throws an exception but other compilers are not (and this ensures the compiler sees it as a the only codepath leaving the clause).

Reply:
The original compiler complaint was that the throw is inside the finally block. Could I replace the "Unreachable code" at the end with this rethrow? I believe the logic will be the same.

Reply:
And what's wrong about a throw from within finally? A finally block is technically just a block of code, like any other. The compiler very likely assumes you're suppressing an exception if you throw from within finally but it's not the case here. 

I don't know if moving that throw will change the logic. Maybe not. Maybe yes. Given the two options, I wouldn't touch it. My concern was that you slipped such things as part of an otherwise "trivial" set of patches.